secrets.sh

```diff
@@ -0,0 +1,219 @@
+This script needs to be run in a cg-deploy-* repo
+
+Operations (only use 1):
+\t-d, --decrypt\tDecrypt secrets
+\t-e, --encrypt\tEncrypt secrets
+\t-c, --check\tDo a healthcheck to make sure the repo is setup correctly
+\t-h, --help\tThis help will print
+
+Options:
+\t-g, --govcloud\tAdd govcloud suffix to all files
+
+Required Environment Variables:
+\tAWS_ACCESS_KEY_ID - Amazon AWS Access Key
+\tAWS_SECRET_ACCESS_KEY - Amazon Secret Access Key
+\tAWS_DEFAULT_REGION - Default Amazon Region
+
+Optional Environment Variables:
+\tCONCOURSE_BUCKET - Bucket with Concourse credentials
+
+Examples:
+
+Decrypt GovCloud secrets:
+\t./secrets.sh -d -g
+
+Encrypt GovCloud secrets:
+\t./secrets.sh -e -g
+"
+
+if [[ $# -lt 1 ]]; then
+  echo -e "${USAGE}"
+  exit 1
+fi
+
+DECRYPT=0
+ENCRYPT=0
+CHECK=0
+GOVCLOUD=0
+
+while [[ $# -gt 0 ]]; do
+  key="$1"
+
+  case ${key} in
+    -d|--decrypt)
+      DECRYPT=1
+      ;;
+    -e|--encrypt)
+      ENCRYPT=1
+      ;;
+    -c|--check)
+      CHECK=1
+      ;;
+    -g|--govcloud)
+      GOVCLOUD=1
+      ;;
+    *)
+      echo "${key} is not even close to a valid option."
+      echo -e "${USAGE}"
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+# Encrypt / Decrypt / Healthcheck only, otherwise exit
+if [ $((${DECRYPT} + ${ENCRYPT} + ${CHECK})) -ne 1 ]; then
+  echo  -e "${USAGE}"
+  exit 1
+fi
+
+DIR=`basename "$PWD"`
+PIPELINE_TASKS="$HOME/.cloudgov/cg-pipeline-tasks"
+
+# Check that we are in a compatible directory
+echo "${DIR}" | egrep '^cg-deploy-[a-zA-Z0-9-]+$' || { echo "${0} is only compatible with cg-deploy-* repos"; exit 1; }
+
+# Check exists? AWS_DEFAULT_REGION, AWS_SECRET_ACCESS_KEY, AWS_ACCESS_KEY_ID
+# http://stackoverflow.com/questions/3601515/how-to-check-if-a-variable-is-set-in-bash
+if [ -z ${AWS_DEFAULT_REGION+x} ]; then
+  echo "AWS_DEFAULT_REGION must be set"
+  exit 1
+fi
+
+if [ -z ${AWS_SECRET_ACCESS_KEY+x} ]; then
+  echo "AWS_SECRET_ACCESS_KEY must be set"
+  exit 1
+fi
+
+if [ -z ${AWS_ACCESS_KEY_ID+x} ]; then
+  echo "AWS_ACCESS_KEY_ID must be set"
+  exit 1
+fi
+
+command -v aws >/dev/null 2>&1 || { echo "AWS CLI required: http://docs.aws.amazon.com/cli/latest/userguide/installing.html"; exit 1; }
+
+if [ ! -d "${PIPELINE_TASKS}/.git" ]; then
+  echo "cg-pipeline-tasks is required to encrypted and decrypt secrets"
+  read -p "We cannot find a git repo, clone it now [Y/N]?" -n 1 -r
+  echo
+  if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+    echo "${PIPELINE_TASKS} cannot be found, aborting."
+    exit 1
+  else
+    mkdir -p ${PIPELINE_TASKS}
+    git clone https://github.com/18F/cg-pipeline-tasks ${PIPELINE_TASKS}
+  fi
+fi
+
+CURRENT_BRANCH=`git rev-parse --abbrev-ref HEAD`
+
+if [ "${CURRENT_BRANCH}" = "master" ]; then
+  TARGET_ENVIRONMENT="production"
+elif [ "${CURRENT_BRANCH}" = "staging" ]; then
+  TARGET_ENVIRONMENT="staging"
+else
+  echo "Secrets currently only exist for production / staging, please"
+  echo "checkout one of those branches and try again."
+  exit 1
+fi
```

Comment:
Hmm, not sure that we have been splitting staging/production configuration with long-lived branches...wondering if this should be a flag instead?